### DISCUSSION TITLE
How to pass parameter --remote-debugging-address to firefox bind 0.0.0.0 inside docker for --start-debugger-server?

### DISCUSSION THREAD

**Discussion by evandrocoan** [jlesage/docker-firefox#246] [Q&A] [answered]
I see 
```
File: docker-firefox/rootfs/startapp.sh
30: /usr/bin/firefox --version
31: exec /usr/bin/firefox "$@" >> /config/log/firefox/output.log 2>> /config/log/firefox/error.log
32: 
```

Can receive arguments, but now to add arguments to startappsh?

**Comment by jlesage** [comment, marked as answer]
Currently there is not built-in mechanism to specify custom arguments to Firefox. I could add support for an environment variable that can be used to set these arguments.

**Reply by evandrocoan** [reply]
Thanks. I think this variable should be useful. Now, I am adding a new layer to the image and editing the startapp.sh with the required arguments.
```
docker run jlesage/firefox
vi /startapp.sh
exit
docker ps
docker commit 7e15b61de473 test
docker stop firefox 
docker rm firefox 
docker run -itd --name=firefox test
```

Note: I see now that --remote-debugging-address is not valid firefox argument, but --start-debugger-server and --remote-debugging-port are.

**Reply by XMethues** [reply]
how did you set the ` --remote-debugging-port`?

**Reply by evandrocoan** [reply]
I just added it to the firefox launch command inside `/startapp.sh` and committed the change as a new docker layer.

**Reply by XMethues** [reply]
I set env FF_CUSTOM_ARGS but it didn't work.

**Reply by juliaisnotavailable** [reply]
I can not get this working with the `- FF_CUSTOM_ARGS=--remote-debugging-port=6000` either. Please can somone advise on what the correct setup is that would allow host to connect to the Firefox debugger. Preferably as a docker compose file.

**Reply by evandrocoan** [reply]
I moved away from using Google Chrome, but I think the problem is the same as Firefox. Both only bind the port --remote-debugging-port to localhost; therefore, you cannot connect to it from outside the docker container. I overcame this problem by using an older version of Chrome, which still binds the port to my docker interface instead of localhost (inside docker, then only available inside docker container with 127.0.0.1). If I upgrade Chrome, I would have to use something inside my docker container to bind a port x like 6001 on my docker network interface IP and forward it to the docker localhost port 6000 (if I used --remote-debugging-port=6000).
